util/chunk/chunk.go

```diff
@@ -267,6 +267,27 @@ func (c *Chunk) CopyConstruct() *Chunk {
 	return newChk
 }
 
+// CopyReconstruct copies this Chunk to dst and removes unselected rows.
+// If dst is nil, it creates a new Chunk and returns it.
+func (c *Chunk) CopyReconstruct(dst *Chunk) *Chunk {
+	if dst == nil {
+		dst = new(Chunk)
+	} else {
+		dst.Reset()
+	}
+	dst.numVirtualRows = c.numVirtualRows
+	dst.capacity = c.capacity
+	dst.sel = c.sel
+	if len(dst.columns) < len(c.columns) {
+		dst.columns = make([]*Column, len(c.columns))
+	}
+
```

Comment:
We should add 
```
else {
    dst.columns = dst.columns[:len(c.columns)]
}
```
Otherwise the method `Chunk.NumCols` is not correct.

Reply:
Update. PTAL @qw4990